core/src/test/java/kafka/admin/ConfigCommandIntegrationTest.java

```diff
@@ -65,219 +74,479 @@
+        EndPoint endpoint = new EndPoint("localhost", 9092,
+                ListenerName.forSecurityProtocol(securityProtocol), securityProtocol);
+        BrokerInfo brokerInfo = BrokerInfo.apply(Broker.apply(id, endpoint,
+                scala.None$.empty()), MetadataVersion.latestTesting(), 9192);
         zkClient.registerBroker(brokerInfo);
     }
 
-    @SafeVarargs
-    static <T> Seq<T> seq(T...seq) {
-        return seq(Arrays.asList(seq));
+    private List<String> generateDefaultAlterOpts(String bootstrapServers) {
+        return asList("--bootstrap-server", bootstrapServers,
+                "--entity-type", "brokers",
+                "--entity-name", "0", "--alter");
+    }
+
+    private void alterAndVerifyConfig(Admin client, Optional<String> brokerId, Map<String, String> config) throws Exception {
+        alterConfigWithKraft(client, brokerId, config);
+        verifyConfig(client, brokerId, config);
+    }
+
+    private void alterConfigWithKraft(Admin client, Optional<String> brokerId, Map<String, String> config) {
+        String configStr = transferConfigMapToString(config);
+        ConfigCommand.ConfigCommandOptions addOpts =
+                new ConfigCommand.ConfigCommandOptions(toArray(alterOpts, entityOp(brokerId), asList("--add-config", configStr)));
+        ConfigCommand.alterConfig(client, addOpts);
     }
 
-    @SuppressWarnings({"deprecation"})
-    static <T> Seq<T> seq(Collection<T> seq) {
-        return JavaConverters.asScalaIteratorConverter(seq.iterator()).asScala().toSeq();
+    private void verifyConfig(Admin client, Optional<String> brokerId, Map<String, String> config) throws Exception {
+        ConfigResource configResource = new ConfigResource(ConfigResource.Type.BROKER, brokerId.orElse(defaultBrokerId));
+        client.describeConfigs(singletonList(configResource))
+                .all()
+                .get()
+                .entrySet()
+                .stream()
+                .filter(entry -> entry.getKey().type() == ConfigResource.Type.BROKER)
+                .map(Map.Entry::getValue)
+                .flatMap(configs -> configs.entries().stream())
+                .map(entry -> entry.name() + "=" + entry.value())
+                .filter(key -> transferConfigMapToString(config).contains(key))
+                .forEach(key -> assertTrue(true, "Config key " + key + " is not found in the broker configs"));
+
+    }
+
+    private void deleteAndVerifyConfig(Admin client, Optional<String> brokerId, Set<String> config) throws Exception {
+        ConfigCommand.ConfigCommandOptions deleteOpts =
+                new ConfigCommand.ConfigCommandOptions(toArray(alterOpts, entityOp(brokerId),
+                        asList("--delete-config", String.join(",", config))));
+        ConfigCommand.alterConfig(client, deleteOpts);
+        verifyConfig(client, brokerId, Collections.emptyMap());
```

Comment:
Could we add a new method to verify the "deleted" config has default value (the config `message.max.bytes` has default value)? for example:
```java
    private void verifyConfig(Admin client, Optional<String> brokerId, Set<String> config) throws Exception {
        ConfigResource configResource = new ConfigResource(ConfigResource.Type.BROKER, brokerId.orElse(defaultBrokerId));
        TestUtils.waitForCondition(() -> {
            Map<String, String> current = client.describeConfigs(singletonList(configResource)).all().get().values()
                .stream().flatMap(e -> e.entries().stream())
                .collect(Collectors.toMap(ConfigEntry::name, ConfigEntry::value));
            return config.stream().allMatch(current::containsKey);
        }, 5000, config + " are not updated");
    }
```